prometheus-to-sd/main.go

```diff
@@ -144,32 +143,18 @@ func main() {
 		glog.Error(http.ListenAndServe(fmt.Sprintf("%s:%d", *debugAddress, *debugPort), expvar.Handler()))
 	}()
 
-	var client *http.Client
-
+	var options []option.ClientOption
 	if *gceTokenURL != "" {
-		client = oauth2.NewClient(context.Background(), config.NewAltTokenSource(*gceTokenURL, *gceTokenBody))
-	} else if *projectOverride != "" {
-		client, err = google.DefaultClient(context.Background(), "https://www.googleapis.com/auth/cloud-platform")
-		if err != nil {
-			glog.Fatalf("Error getting default credentials: %v", err)
-		}
-		glog.Infof("Created a client with the default credentials")
-	} else {
-		ts, err := google.DefaultTokenSource(context.Background(), "https://www.googleapis.com/auth/cloud-platform")
-		if err != nil {
-			glog.Fatalf("Error creating default token source: %v", err)
-		}
-		client = oauth2.NewClient(context.Background(), ts)
+		ts := config.NewAltTokenSource(*gceTokenURL, *gceTokenBody)
+		options = append(options, option.WithTokenSource(ts))
```

Comment:
> *projectOverride != ""
I find this else is deleted. Will it respect projectOverride?

Reply:
Yes, the project name is in the request body,  see https://github.com/GoogleCloudPlatform/k8s-stackdriver/pull/742/files#diff-fabb01b5a0dd10f9e48b220cf31c93d51c6fd5a5516b4254ca63232c43a0dfe8R61

Reply:
+1 to @linxiulei's comment. if you trace what's being done by [google.DefaultClient()](https://cs.opensource.google/go/x/oauth2/+/ee480838109b20d468babcb00b7027c82f962065:google/default.go;l=73) in the `*projectOverride != ""` case, it's exactly the same as the other case. The projectOverride string is not used at all.